src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -247,25 +247,21 @@ struct
   (* Basic transfer functions. *)
   let assign ctx (lv:lval) e =
     let st = ctx.local in
-    if !AnalysisState.global_initialization && e = MyCFG.unknown_exp then
-      st (* ignore extern inits because there's no body before assign, so env is empty... *)
```

Comment:
It seems strange why this was a problem earlier, but isn't a problem now. Did you test that this does not cause us to have top everywhere, i.e., also for other variables that get a proper initialization later?

Also what is the `env` here now?

Reply:
It's hard to git blame this due to some massive changes at the affine equalities time, but I suspect it might be some very old relic. Because global initializers are similar (there's no `body` to explicitly add them all) but there's no problem with those.
Nowadays, `write_global` just adds the assigned variable to the environment: https://github.com/goblint/analyzer/blob/2fa4f55e682da3ad937e394fe20449eeef4c9634/src/analyses/apron/relationAnalysis.apron.ml#L136-L139

This might be a remainder from the days when our Apron analysis didn't support globals at all.